### PR TITLE
Add fhirpath supplements

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes #https://github.com/google/android-fhir/issues/2163

**Description**
Support fhirpath supplement %questionnaire, and %qItem literal. This allow to get current questionnaire and item into fhirpath expressions on questionnaire items when doing evaluation. Details for spec defined here https://build.fhir.org/ig/HL7/sdc/expressions.html#fhirpath-supplements .

**Type**
Choose one: Feature

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/fhirpath/ExpressionEvaluator.kt [hunk 1]
```diff
@@ -425,13 +436,11 @@ internal class ExpressionEvaluator(
...
-          /* appContext= */ dependentVariables,
-          /* focusResource= */ questionnaireResponse,
-          /* rootResource= */ null,
-          /* base= */ null,
-          /* path= */ expression.expression,
+      evaluateToBase(
+          questionnaireResponse,
+          null,
+          expression.expression,
+          dependentVariables,
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/fhirpath/ExpressionEvaluator.kt:443]: add named parameters

**Review by jingtang10 — CHANGES_REQUESTED** [review]
hi maimoona - thanks for this great work!

**Comment by jingtang10** [comment]
i don't hink this pr builds with the latest changes. can you fix please @maimoonak? thanks

**Review by jingtang10 — APPROVED** [review]
Please double-check the code paths going through `evaluateExpression` and `evaluateVariable` and see if there's any more unification you can do.

Thanks for this PR! Great job!

**Comment by maimoonak** [comment]
> Please double-check the code paths going through `evaluateExpression` and `evaluateVariable` and see if there's any more unification you can do.
> 
> Thanks for this PR! Great job!

https://github.com/google/android-fhir/issues/2373 The methods are covered by a good amount of test which indeed is important to ensure complex variable evaluation validation. We need to figureout a way to keep unit tests. (VisibleForTesting annotation not working)

**Pull request merged**